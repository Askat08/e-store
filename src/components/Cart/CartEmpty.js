import React from "react";

export default function CartEmpty() {
  return (
    <div className="conainer m-5">
      <div className="row">
        <div className="col-10 mx-auto text-center">
          <h1>Cart is empty</h1>
        </div>
      </div>
    </div>
  );
}
