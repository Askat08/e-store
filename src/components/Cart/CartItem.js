import React from "react";

export default function CartItem({ props, item }) {
  const { id, title } = item;
  const { path, extension } = item.thumbnail;
  const price = item.prices[0].price;
  const quantity = item.count;
  const https = Array.from(path);
  https.splice(4, 0, "s");
  return (
    <div className="row my-2 text-capitilize text-center">
      <div className="col-10 mx-auto col-lg-2">
        <img
          src={https.join("") + "." + extension}
          alt={title}
          className="img-fluid"
          style={{ width: "5rem", height: "5rem" }}
        />
      </div>
      <div className="col-10 mx-auto col-lg-2">
        <span className="d-lg-none">Comic : </span> {title}
      </div>
      <div className="col-10 mx-auto col-lg-2">
        <span className="d-lg-none">Price : </span>
        <strong>{`$${price.toFixed(2)}`}</strong>
      </div>
      <div className="col-10 mx-auto col-lg-2 my-2 my-lg-0">
        <div className="d-flex justify-content-center">
          <div className="">
            <span
              className="btn btn-cart mx-1"
              onClick={() => props.decrementCount(id)}
            >
              -
            </span>
            <span className="btn btn-cart mx-1">{quantity}</span>
            <span
              className="btn btn-cart mx-1"
              onClick={() => props.incrementCount(id)}
            >
              +
            </span>
          </div>
        </div>
      </div>

      <div className="col-10 mx-auto col-lg-2">
        <div
          style={{ cursor: "pointer" }}
          onClick={() => {
            props.removeItemFromCart(id, props.cart);
          }}
        >
          <i className="fas fa-trash text-danger" />
        </div>
      </div>

      <div className="col-10 mx-auto col-lg-2">
        <strong>Item total : $ {(price * quantity).toFixed(2)}</strong>
      </div>
    </div>
  );
}
