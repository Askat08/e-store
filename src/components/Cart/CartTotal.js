import React from "react";
import { Link } from "react-router-dom";

export default function CartTotal({ props }) {
  const { cartTax, cartTotal, cartSubTotal, handleClearCart } = props;

  return (
    <React.Fragment>
      <div className="container">
        <div className="row">
          <div className="col-10 mt-2 ml-sm-5 ml-md-auto col-sm-8 text-right text-capitilize">
            <Link to="/">
              <button
                className="btn btn-outline-danger text-uppercase mb-3 px-5"
                type="button"
                onClick={() => handleClearCart()}
              >
                clear cart
              </button>
            </Link>
            <h5>
              <span className="text-info">subtotal : </span>
              <strong>$ {cartSubTotal}</strong>
            </h5>
            <h5>
              <span className="text-info">tax* : </span>
              <strong>$ {cartTax}</strong>
            </h5>
            <h5>
              <span className="text-info">total : </span>
              <strong>$ {cartTotal}</strong>
            </h5>
            <br />
            <br />
            <br />
            <br />
            <p>*New York State Sales Tax 8.875%</p>
          </div>
        </div>
      </div>
    </React.Fragment>
  );
}
