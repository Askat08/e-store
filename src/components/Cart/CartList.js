import React from "react";
import CartItem from "./CartItem";

export default function CartList({ cart, props }) {
  const filteredCart = cart.filter((v, i, a) => a.indexOf(v) === i);
  return (
    <div className="container-fluid">
      {filteredCart.map((item) => {
        return <CartItem key={item.id} item={item} props={props} />;
      })}
    </div>
  );
}
