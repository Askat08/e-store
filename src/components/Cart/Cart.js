import React from "react";
import CartData from "./CartData";
import CartEmpty from "./CartEmpty";
import CartList from "./CartList";
import CartTotal from "./CartTotal";

const Cart = (props) => {
  if (props.cart.length > 0) {
    const { cart } = props;
    return (
      <div>
        <h3 className="m-5 text-center">Comics in cart</h3>
        <CartData />
        <CartList cart={cart} props={props} />
        <CartTotal props={props} />
      </div>
    );
  } else {
    return <CartEmpty />;
  }
};

export default Cart;
