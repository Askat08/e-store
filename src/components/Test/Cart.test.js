import React from "react";
import { shallow } from "enzyme";
import Cart from "../Cart/Cart";

const propsMock = {
  cart: [
    { title: "thor", id: 1, count: 1, prices: [{ price: 2.99 }] },
    { title: "loki", id: 2, count: 1, prices: [{ price: 2.25 }] },
    { title: "hulk", id: 3, count: 1, prices: [{ price: 23.99 }] },
    { title: "spider-man", id: 4, count: 1, prices: [{ price: 15.19 }] },
  ],
};

describe("Cart component test", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Cart {...propsMock} />);
  });

  it("shallow Cart component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("h3 has proper classname and text", () => {
    expect(wrapper.find("h3").hasClass("m-5 text-center")).toBeTruthy();
    expect(wrapper.find("h3").text()).toEqual("Comics in cart");
  });
});
