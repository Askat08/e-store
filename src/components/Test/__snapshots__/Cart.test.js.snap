// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component test shallow Cart component 1`] = `
<div>
  <h3
    className="m-5 text-center"
  >
    Comics in cart
  </h3>
  <CartData />
  <CartList
    cart={
      Array [
        Object {
          "count": 1,
          "id": 1,
          "prices": Array [
            Object {
              "price": 2.99,
            },
          ],
          "title": "thor",
        },
        Object {
          "count": 1,
          "id": 2,
          "prices": Array [
            Object {
              "price": 2.25,
            },
          ],
          "title": "loki",
        },
        Object {
          "count": 1,
          "id": 3,
          "prices": Array [
            Object {
              "price": 23.99,
            },
          ],
          "title": "hulk",
        },
        Object {
          "count": 1,
          "id": 4,
          "prices": Array [
            Object {
              "price": 15.19,
            },
          ],
          "title": "spider-man",
        },
      ]
    }
    props={
      Object {
        "cart": Array [
          Object {
            "count": 1,
            "id": 1,
            "prices": Array [
              Object {
                "price": 2.99,
              },
            ],
            "title": "thor",
          },
          Object {
            "count": 1,
            "id": 2,
            "prices": Array [
              Object {
                "price": 2.25,
              },
            ],
            "title": "loki",
          },
          Object {
            "count": 1,
            "id": 3,
            "prices": Array [
              Object {
                "price": 23.99,
              },
            ],
            "title": "hulk",
          },
          Object {
            "count": 1,
            "id": 4,
            "prices": Array [
              Object {
                "price": 15.19,
              },
            ],
            "title": "spider-man",
          },
        ],
      }
    }
  />
  <CartTotal
    props={
      Object {
        "cart": Array [
          Object {
            "count": 1,
            "id": 1,
            "prices": Array [
              Object {
                "price": 2.99,
              },
            ],
            "title": "thor",
          },
          Object {
            "count": 1,
            "id": 2,
            "prices": Array [
              Object {
                "price": 2.25,
              },
            ],
            "title": "loki",
          },
          Object {
            "count": 1,
            "id": 3,
            "prices": Array [
              Object {
                "price": 23.99,
              },
            ],
            "title": "hulk",
          },
          Object {
            "count": 1,
            "id": 4,
            "prices": Array [
              Object {
                "price": 15.19,
              },
            ],
            "title": "spider-man",
          },
        ],
      }
    }
  />
</div>
`;
